Sources/StytchUI/AuthInputViewController.swift

```diff
@@ -140,32 +140,53 @@ final class AuthInputViewController: BaseViewController<AuthInputState, AuthInpu
         continueButton.isEnabled = isCurrentInputValid
     }
 
+    private func launchMagicLinkPassword(email: String) async throws {
+        let intent = try await viewModel.getUserIntent(email: email)
+        if let intent = intent {
+            DispatchQueue.main.async {
+                self.launchPassword(intent: intent, email: email, magicLinksEnabled: self.viewModel.state.config.magicLink != nil)
+            }
+        } else {
+            try await viewModel.resetPassword(email: email)
+            DispatchQueue.main.async {
+                self.launchCheckYourEmailResetReturning(email: email)
+            }
+        }
+    }
+
+    private func launchMagicLinkOnly(email: String) async throws {
+        try await viewModel.sendMagicLink(email: email)
+        DispatchQueue.main.async {
+            self.launchCheckYourEmail(email: email)
+        }
+    }
+
+    private func launchPasswordOnly(email: String) async throws {
+        let intent = try await viewModel.getUserIntent(email: email)
+        if let intent = intent {
```

Comment:
Should there be an else here, or is it enforced that there will always be an intent for this method?

Reply:
Yes, there should always be an intent here